src/WebJobs.Extensions/Extensions/Timers/Scheduling/TimerSchedule.cs

```diff
@@ -98,5 +94,41 @@ internal static TimerSchedule Create(TimerTriggerAttribute attribute, INameResol
 
             return schedule;
         }
+
+        /// <summary>
+        /// For schedules with an occurrence more frequent than 1 minute, we disable schedule monitoring.
+        /// </summary>
+        /// <param name="cronSchedule">The cron schedule to check.</param>
+        /// <returns>True if monitoring should be disabled, false otherwise.</returns>
+        internal static bool ShouldDisableScheduleMonitor(CronSchedule cronSchedule, DateTime now)
+        {
+            // take the original expression minus the seconds portion
+            string expression = cronSchedule.InnerSchedule.ToString();
+            var expressions = expression.Split(' ');
+
+            // If any of the minute or higher fields contain non-wildcard expressions
+            // the schedule can be longer than 1 minute. I.e. the only way for all occurrences
+            // to be less than or equal to a minute is if all these fields are wild ("* * * * *").
+            bool hasNonSecondRestrictions = !expressions.Skip(1).All(p => p == "*");
+
+            if (!hasNonSecondRestrictions)
+            {
+                // If to here, we know we're dealing with a schedule of the form X * * * * *
+                // so we just need to consider the seconds expression to determine if it occurs
+                // more frequently than 1 minute.
+                // E.g. an expression like */10 * * * * * occurs every 10 seconds, while an
+                // expression like 0 * * * * * occurs exactly once per minute.
+                DateTime[] nextOccurrences = cronSchedule.InnerSchedule.GetNextOccurrences(now, now + TimeSpan.FromMinutes(1)).ToArray();
+                if (nextOccurrences.Length > 1)
+                {
+                    // if there is more than one occurrence due in the next minute,
+                    // assume that this is a sub-minute constant schedule and disable
+                    // persistence
+                    return true;
+                }
+            }
+
+            return false;
+        }
```

Comment:
It may be worth writing out (debug-level) logs here to help us diagnose any problems. Something that tells us if we enabled/disabled the monitor, and why.

Reply:
Added a debug log. The fact that our disable logic kicked in is sufficient - we log the schedule details elsewhere.